### PR TITLE
Added C++ fundamental book

### PR DESCRIPTION
**Pull request by vineetsingh0809**
What does this PR do?
Add resource(s)

For resources
Description
Added an ebook of Fundamental of C++ programming

Why is this valuable (or not)?
Because this helps in preparing for further studies

How do we know it's really free?
I have attached a free link to the ebook

For book lists, is it a book? For course lists, is it a course? etc.
Book

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md
Comment (LuigiImVector) [inline books/free-programming-books-langs.md:499]:
```suggestion
* [Fundamentals of C++ Programming](https://ia800701.us.archive.org/34/items/2018FundamentalsOfCppProgramming/2018_fundamentals-of-cpp-programming.pdf) - Richard L. Halterman (PDF) (:construction: *in process*) *(:card_file_box: archived)*
```

**Comment by eshellman** [comment]
The book is both archived and in process?   When we started the "in process" label, we hoped to periodically review the titles and remove the ones that had been abandoned.

**Comment by LuigiImVector** [comment]
I found some old versions of the book and I thought there could be more in the future even if it is hosted on archive.org, to avoid misunderstandings I delete the tag "in progress".

**Comment by eshellman** [comment]
appears to have been put on archive.org by author. We should use https://archive.org/details/2018FundamentalsOfCppProgramming

**Comment by LuigiImVector** [comment]
> 
> 
> appears to have been put on archive.org by author. We should use https://archive.org/details/2018FundamentalsOfCppProgramming

I have not inserted that link because the links in the testpo do not work

**Comment by davorpa** [comment]
My 2 cents... And why don't ask to @halterman directly about? Maybe he has a copy that can upload to the Book exercises repo: https://github.com/halterman/CppBook-SourceCode

3th party consolidated url: https://tfetimes.com/wp-content/uploads/2015/04/progcpp.pdf

**Review by LuigiImVector — COMMENTED** [review]
We wait for an answer from the author of the book

**Comment by eshellman** [comment]
> > appears to have been put on archive.org by author. We should use https://archive.org/details/2018FundamentalsOfCppProgramming
> 
> I have not inserted that link because the links in the testpo do not work

Not sure I understand. what is "testpo"?

**Comment by LuigiImVector** [comment]
> 
> Not sure I understand. what is "testpo"?

sorry, typo error. [In this link](https://archive.org/details/2018FundamentalsOfCppProgramming) the hyperlinks doesn't work (see page 10-11, for example)

**Comment by davorpa** [comment]
> sorry, typo error. [In this link](https://archive.org/details/2018FundamentalsOfCppProgramming) the hyperlinks doesn't work (see page 10-11, for example)

I catch up! Only works wrong on the Viewer. On download links are ok. More over...

image

> latest backup **(July 11, 2019)**: http://web.archive.org/web/20191005170118/https://python.cs.southern.edu/cppbook/progcpp.pdf

I will opt with this above since is the most recent :wink: Are we all in joint tune?

**Comment by LuigiImVector** [comment]
> In anyway, problem is reported. I will opt with this above since is the most recent 😉
> 
> Are we all in joint tune?

Alright, maybe before merging we wait for @halterman's answer

**Comment by eshellman** [comment]
we can proceed, I think

File: books/free-programming-books-langs.md
Comment (davorpa) [inline books/free-programming-books-langs.md:499]: Use latest archived available version
```suggestion
* [Fundamentals of C++ Programming](https://web.archive.org/web/20191005170118/https://python.cs.southern.edu/cppbook/progcpp.pdf) - Richard L. Halterman (PDF) *(:card_file_box: archived)*
```

**Review by davorpa — APPROVED** [review]
LGTM :+1:

**Pull request merged**
